pkg/monitor/intervalcreation/node.go

```diff
@@ -100,3 +114,185 @@ func IntervalsFromEvents_NodeChanges(events monitorapi.Intervals, _ monitorapi.R
 
 	return intervals
 }
+
+func IntervalsFromNodeLogs(ctx context.Context, kubeClient kubernetes.Interface, beginning, end time.Time) (monitorapi.Intervals, error) {
+	ret := monitorapi.Intervals{}
+
+	allNodes, err := kubeClient.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	lock := sync.Mutex{}
+	errCh := make(chan error, len(allNodes.Items))
+	wg := sync.WaitGroup{}
+	for _, node := range allNodes.Items {
+		wg.Add(1)
+		go func(ctx context.Context, nodeName string) {
+			defer wg.Done()
+
+			// TODO limit by begin/end here instead of post-processing
+			nodeLogs, err := nodedetails.GetNodeLog(ctx, kubeClient, nodeName, "kubelet")
+			if err != nil {
+				errCh <- err
+				return
+			}
+			newEvents := eventsFromKubeletLogs(nodeName, nodeLogs)
+
+			lock.Lock()
+			defer lock.Unlock()
+			ret = append(ret, newEvents...)
+		}(ctx, node.Name)
+	}
+	wg.Wait()
+
+	errs := []error{}
+	for len(errCh) > 0 {
+		err := <-errCh
+		errs = append(errs, err)
+	}
+
+	return ret, utilerrors.NewAggregate(errs)
+}
+
+func eventsFromKubeletLogs(nodeName string, kubeletLog []byte) monitorapi.Intervals {
+	ret := monitorapi.Intervals{}
+
+	scanner := bufio.NewScanner(bytes.NewBuffer(kubeletLog))
+	for scanner.Scan() {
+		currLine := scanner.Text()
+		ret = append(ret, readinessFailure(currLine)...)
+		ret = append(ret, readinessError(currLine)...)
+	}
+
+	return ret
```

Comment:
Suspicion here that there are a few things that could go wrong that are getting glazed over in this function and below. Should we handle errors here, at least outputting something that we could find in search.ci if suddenly we couldn't parse these the way we'd expect? (I'm ok with it just proceeding, just looking for something in build log we can find)

Reply:
is it better to not render the problem at all or render it at an unusual time so someone notices.  choosing zero will destroy the chart.  Choosing now will look weird, but will not time-shift the entire chart.

Reply:
I guess I'd have expected logging the problematic lines and then dropping the interval. If the goal is to still display it at a weird point and assume that is more visible to someone looking, I can see that potentially helping. I would still log the problem though so we've got something definite we can look for. It somewhat depends what these particular lines are or could be.